File: token-client-core/src/main/java/no/nav/security/token/support/client/core/OAuth2GrantType.java
package no.nav.security.token.support.client.core;

import java.util.Optional;

public record OAuth2GrantType(String value) {
    public static final OAuth2GrantType JWT_BEARER = new OAuth2GrantType("urn:ietf:params:oauth:grant-type:jwt-bearer");
    public static final OAuth2GrantType CLIENT_CREDENTIALS = new OAuth2GrantType("client_credentials");
    public static final OAuth2GrantType TOKEN_EXCHANGE = new OAuth2GrantType("urn:ietf:params:oauth:grant-type:token-exchange");

    public OAuth2GrantType(String value) {
        this.value = Optional.ofNullable(value)
            .orElseThrow(() -> new OAuth2ClientException("value for OAuth2GrantType cannot be null"));
    }
}

File: token-client-core/src/main/kotlin/no/nav/security/token/support/client/core/OAuth2GrantType.kt
package no.nav.security.token.support.client.core

import com.nimbusds.oauth2.sdk.GrantType
import kotlin.DeprecationLevel.WARNING

@Deprecated("Use GrantType from nimbus instead", ReplaceWith("GrantType"), WARNING)
data class OAuth2GrantType(@JvmField val value : String) {
     fun value() = value

    companion object {
       @JvmField
       @Deprecated("Use com.nimbusds.oauth2.sdk.GrantType instead", ReplaceWith("GrantType.JWT_BEARER"), WARNING)
       val JWT_BEARER = GrantType(GrantType.JWT_BEARER.value)
        @JvmField
        @Deprecated("Use com.nimbusds.oauth2.sdk.GrantType instead", ReplaceWith("GrantType.CLIENT_CREDENTIALS"), WARNING)
        val CLIENT_CREDENTIALS = GrantType(GrantType.CLIENT_CREDENTIALS.value)
        @JvmField
        @Deprecated("Use com.nimbusds.oauth2.sdk.GrantType instead", ReplaceWith("GrantType.TOKEN_EXCHANGE"), WARNING)
        val TOKEN_EXCHANGE = GrantType(GrantType.TOKEN_EXCHANGE.value)
    }
}